src/Enqueue/LocalizationConstants.php

```diff
@@ -0,0 +1,430 @@
+	 * Create a new admin instance.
+	 *
+	 * @param ActiveRouteInterface $dynamicsCrmRoute          Dynamics CRM route object which holds values we need to localize.
+	 * @param ActiveRouteInterface $buckarooIdealRoute        Buckaroo (Ideal) route object which holds values we need to localize.
+	 * @param ActiveRouteInterface $buckarooEmandateRoute     Buckaroo (Emandate) route object which holds values we need to localize.
+	 * @param ActiveRouteInterface $buckarooPayByEmailRoute Buckaroo (Pay By Email) route object which holds values we need to localize.
+	 * @param ActiveRouteInterface $sendEmailRoute            Send Email route object which holds values we need to localize.
+	 * @param ActiveRouteInterface $mailchimpRoute             Mailchimp route object which holds values we need to localize.
+	 * @param Mailchimp            $mailchimp                   Mailchimp implementation.
+	 * @param ActiveRouteInterface $mailerliteRoute            Mailerlite route object which holds values we need to localize.
+	 * @param Mailerlite           $mailerlite                  Mailerlite implementation.
+	 */
+	public function __construct(
+		ActiveRouteInterface $dynamicsCrmRoute,
+		ActiveRouteInterface $buckarooIdealRoute,
+		ActiveRouteInterface $buckarooEmandateRoute,
+		ActiveRouteInterface $buckarooPayByEmailRoute,
+		ActiveRouteInterface $sendEmailRoute,
+		ActiveRouteInterface $mailchimpRoute,
+		Mailchimp $mailchimp,
+		ActiveRouteInterface $mailerliteRoute,
+		Mailerlite $mailerlite
+	) {
+		$this->dynamicsCrmRoute = $dynamicsCrmRoute;
+		$this->buckarooIdealRoute = $buckarooIdealRoute;
+		$this->buckarooEmandateRoute = $buckarooEmandateRoute;
+		$this->buckarooPayByEmailRoute = $buckarooPayByEmailRoute;
+		$this->sendEmailRoute = $sendEmailRoute;
+		$this->mailchimpRoute = $mailchimpRoute;
+		$this->mailchimp = $mailchimp;
+		$this->mailerliteRoute = $mailerliteRoute;
+		$this->mailerlite = $mailerlite;
+	}
+
+	/**
+	 * Define all variables we need in both editor and frontend.
+	 *
+	 * @return array
+	 */
+	public function getLocalizations(): array
+	{
+		$localization = [
+			self::LOCALIZATION_KEY => [
+				'siteUrl'           => get_site_url(),
+				'isDynamicsCrmUsed' => has_filter(Filters::DYNAMICS_CRM),
+				'isBuckarooUsed'    => has_filter(Filters::BUCKAROO),
+				'isMailchimpUsed'   => has_filter(Filters::MAILCHIMP),
+				'isMailerliteUsed'  => has_filter(Filters::MAILERLITE),
+				'hasThemes'         => has_filter(Filters::GENERAL),
+				'content' => [
+					'formLoading' => esc_html__('Form is submitting, please wait.', 'eightshift-forms'),
+					'formSuccess' => esc_html__('Form successfully submitted.', 'eightshift-forms'),
+				],
+				'sendEmail' => [
+					'restUri' => $this->sendEmailRoute->getRouteUri(),
+				],
+				'internalServerErrorMessage' => esc_html__('Internal server error', 'eightshift-forms'),
+			],
+		];
+
+		if (has_filter(Filters::GENERAL)) {
+			$localization = $this->addGeneralConstants($localization);
```

Comment:
Is this intended behavior? You are overwriting the above `$localization` variable. 

Whenever you have one of these filters, the above localization will be overwritten. Or did you want to push to the `$localization` array? If that's the case, you missed `[]` at the end of the `$localization` variable.

Reply:
Yes, this is intended behavior. Reasoning is that it's more pretty in the main `getLocalizations()` function (no need to write down all the keys in it) and then you can view specific constants for each integration in their respective functions.

Reply:
Those functions always append localizations to the required keys, they never overwrite it.